src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -221,7 +222,9 @@ public Builder addMember(String name, CodeBlock codeBlock) {
      * depending on the given {@code value} object. Falls back to {@code "$L"} literal format if
      * the class of the given {@code value} object is not supported.
      */
-    private Builder addValue(String memberName, Object value) {
+    public Builder addMemberForValue(String memberName, Object value) {
```

Comment:
You do remember the overload clash with the single method name? The extra
name was on purpose.
Or are you suggesting dropping one of the old methods?

On Sat, Jan 16, 2016 at 8:03 PM, Jesse Wilson <notifications@github.com>
wrote:

> In src/main/java/com/squareup/javapoet/AnnotationSpec.java
> <https://github.com/square/javapoet/pull/403#discussion_r49935756>:
>
> > @@ -221,7 +222,9 @@ public Builder addMember(String name, CodeBlock codeBlock) {
> >       * depending on the given {@code value} object. Falls back to {@code "$L"} literal format if
> >       * the class of the given {@code value} object is not supported.
> >       */
> > -    private Builder addValue(String memberName, Object value) {
> > +    public Builder addMemberForValue(String memberName, Object value) {
>
> Hmmm . . . . is our approach wrong here?
>
> With this API we offer two ways to do everything:
>
>   builder.addMember("foo", "$S", "hello");
>   builder.addMemberForValue("foo", "hello");
>   builder.addMember("bar", "$T.$N", Roshambo.class, Roshambo.ROCK);
>   builder.addMemberForValue("bar", Roshambo.ROCK);
>
> Is it better to permit a single API?
>
>   builder.addMember("foo", "$S", "hello");
>   builder.addMember("foo", "$V", "hello");
>   builder.addMember("bar", "$T.$N", Roshambo.class, Roshambo.ROCK);
>   builder.addMember("bar", "$V", Roshambo.ROCK);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/javapoet/pull/403/files#r49935756>.
>